samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -19,258 +19,212 @@
+import org.apache.samza.serializers.Serde
+import org.apache.samza.metrics.MetricsRegistry
 import org.apache.samza.serializers.CheckpointSerde
-import org.apache.samza.system.SystemStreamMetadata.SystemStreamPartitionMetadata
-import org.apache.samza.system.kafka.{KafkaSystemAdmin, KafkaStreamSpec}
-import org.apache.samza.system.{StreamSpec, SystemAdmin, _}
+import org.apache.samza.system.kafka.KafkaStreamSpec
+import org.apache.samza.system._
 import org.apache.samza.util._
 import org.apache.samza.{Partition, SamzaException}
 
-import scala.collection.JavaConversions._
 import scala.collection.mutable
 
 /**
- * Kafka checkpoint manager is used to store checkpoints in a Kafka topic.
- * To read a checkpoint for a specific taskName, we find the newest message
- * keyed to that taskName. If there is no such message, no checkpoint data
- * exists.  The underlying log has a single partition into which all
- * checkpoints and TaskName to changelog partition mappings are written.
- *
- * This class is thread safe for writing but not for reading checkpoints.
- * This is currently OK since checkpoints are only read on the main thread.
- */
-class KafkaCheckpointManager(
-                              clientId: String,
-                              checkpointTopic: String,
-                              val systemName: String,
-                              replicationFactor: Int,
-                              socketTimeout: Int,
-                              bufferSize: Int,
-                              fetchSize: Int,
-                              getSystemConsumer: () => SystemConsumer,
-                              getSystemAdmin: () => SystemAdmin,
-                              val metadataStore: TopicMetadataStore,
-                              getSystemProducer: () => SystemProducer,
-                              val connectZk: () => ZkUtils,
-                              systemStreamPartitionGrouperFactoryString: String,
-                              failOnCheckpointValidation: Boolean,
-                              val retryBackoff: ExponentialSleepStrategy = new ExponentialSleepStrategy,
-                              serde: CheckpointSerde = new CheckpointSerde,
-                              checkpointTopicProperties: Properties = new Properties) extends CheckpointManager with Logging {
+  * A [[CheckpointManager]] that uses a Kafka topic-partition to store the [[Checkpoint]] corresponding to
+  * a task.
+  *
+  * <p> The Kafka partition provides an abstraction of a log to which all [[Checkpoint]]s are appended to. The
+  * checkpoints written to the log are keyed by their corresponding taskName.
+  *
+  * <p> This class is thread safe for writing but not for reading checkpoints. This is currently OK since checkpoints
+  * are only read on the main thread.
+  */
+class KafkaCheckpointManager(checkpointSpec: KafkaStreamSpec,
+  systemFactory: SystemFactory,
+  failOnCheckpointValidation: Boolean,
```

Comment:
This is not a great field name. Maybe rename everything upstream up until the config to "validateCheckpoints" or "failOnInvalidCheckpoints"?